datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -902,176 +902,394 @@ pub fn array_positions(args: &[ArrayRef]) -> Result<ArrayRef> {
+                                el.clone().unwrap()
+                            }
+                        })
+                        .collect::<Vec<_>>();
+
+                    let mut i: i32 = 0;
+                    let mut replaced_offsets = vec![i];
+                    replaced_offsets.extend(
+                        replaced_vec
+                            .clone()
+                            .into_iter()
+                            .map(|a| {
+                                i += a.len() as i32;
+                                i
+                            })
+                            .collect::<Vec<_>>(),
+                    );
+
+                    let mut replaced_values = downcast_arg!(
+                        new_empty_array(&from_array.value_type()),
+                        $ARRAY_TYPE
+                    )
+                    .clone();
+                    for replaced_list in replaced_vec {
+                        replaced_values = downcast_arg!(
+                            compute::concat(&[&replaced_values, &replaced_list])?,
+                            $ARRAY_TYPE
+                        )
+                        .clone();
                     }
+
+                    let field = Arc::new(Field::new(
+                        "item",
+                        from_array.value_type().clone(),
+                        true,
+                    ));
+                    let replaced_array = ListArray::try_new(
+                        field,
+                        OffsetBuffer::new(replaced_offsets.clone().into()),
+                        Arc::new(replaced_values),
+                        None,
+                    )?;
+
+                    values = downcast_arg!(
+                        compute::concat(&[&values, &replaced_array,])?.clone(),
+                        ListArray
+                    )
+                    .clone();
+                    offsets.push(last_offset + replaced_array.len() as i32);
+                }
+                None => {
+                    offsets.push(last_offset);
                 }
-                None => builder.append_null(),
             }
         }
-        let arr = builder.finish();
 
-        let mut scalars = vec![];
-        for i in 0..arr.len() {
-            scalars.push(ColumnarValue::Scalar(ScalarValue::try_from_array(&arr, i)?));
-        }
-        scalars
+        let field = Arc::new(Field::new("item", $FROM.data_type().clone(), true));
+
+        Arc::new(ListArray::try_new(
+            field,
+            OffsetBuffer::new(offsets.into()),
+            Arc::new(values),
+            None,
+        )?)
     }};
 }
 
 /// Array_replace SQL function
-pub fn array_replace(args: &[ColumnarValue]) -> Result<ColumnarValue> {
-    let arr = match &args[0] {
-        ColumnarValue::Scalar(scalar) => scalar.to_array().clone(),
-        ColumnarValue::Array(arr) => arr.clone(),
-    };
+pub fn array_replace(args: &[ArrayRef]) -> Result<ArrayRef> {
```

Comment:
I think `array_replace(array, from, to, max)` should be `array_replaces` since not replace only one. `array_remove(array, element, max)` should be `array_remove(array, elements, max)` since we might remove elements, the same as `array_removes(array, elements)`

Reply:
I think `replace` and `remove` are helpful functions that are nice to have.